controller/server/controller.go

```diff
@@ -261,6 +262,9 @@ func (c *Controller) Run() {
 				return
 			}
 
+			//after request context is filled so that api session is present for session expiration headers
+			response.AddHeaders(rc)
```

Comment:
This location is the root of the HTTP server handler delegation. Setting the headers here means unless downstream code unsets or resets it, these headers will appear on every response. This is an improvement as previously headers were set in multiple locations. 

The only additional place that these headers need to be set is during authentication when the ApiSession used for expiration headers comes into existence after this code runs.